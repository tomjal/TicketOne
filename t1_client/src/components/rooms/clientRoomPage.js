import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { getClientRoom } from './../../actions/roomsActionCreators';
import { getMessagesByRoom, sendMessageToRoom } from './../../actions/messagesActionCreators';

import Room from './room';

const mapStateToProps = (state) => {
  return {
    id: state.context.id,
    role: state.context.role,
    messages: state.messages
  }
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getClientRoom,
    getMessagesByRoom,
    sendMessageToRoom
  }, dispatch);
};

export class ClientRoomPage extends Component {
  constructor() {
    super();
    this.sendMessageCallback = this.sendMessageCallback.bind(this);
  }
  componentDidMount() {
    this.props.getClientRoom();
    this.props.getMessagesByRoom(this.props.id);
  }
  sendMessageCallback(message, id) {
    this.props.sendMessageToRoom(message, id, this.props.role, this.props.id);
  }
  render() {
    const { id, role, messages } = this.props;
    return (
      <div className="room-flex-container">
        <Room
          id={id}
          role={role}
          sendMessageCallback={this.sendMessageCallback}
          messages={messages} />
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClientRoomPage);
