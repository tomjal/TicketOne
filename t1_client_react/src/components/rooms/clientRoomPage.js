import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { createClientRoom, getClientRooms, markRoomAsResolved, markRoomAsUnresolved } from './../../actions/roomsActionCreators';
import { sendMessageToRoom } from './../../actions/messagesActionCreators';
import { getClientInitRoomsData } from './../../actions/combinedActionCreators';

import RoomWidget from './roomWidget';
import CreateNewRoomWidget from './createNewRoomWidget';

const mapStateToProps = (state) => {
  return {
    id: state.context.id,
    role: state.context.role,
    messages: state.messages,
    rooms: state.rooms
  }
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    createClientRoom,
    getClientRooms,
    sendMessageToRoom,
    markRoomAsResolved,
    markRoomAsUnresolved,
    getClientInitRoomsData
  }, dispatch);
};

export class ClientRoomPage extends Component {
  componentDidMount() {
    //combined init data client
    this.props.getClientInitRoomsData(this.props.id);
  }
  sendMessageCallback = (message, roomId) => {
    this.props.sendMessageToRoom(message, roomId, this.props.role, this.props.id);
  }
  createRoomCallback = (roomTopic) => {
    this.props.createClientRoom(this.props.id, roomTopic);
  }
  render() {
    const { id, role, messages, rooms } = this.props;
    return (
      <div className="room-flex-container">
        {rooms.map((room, i) => {
          let localMessages = [];
          if (messages[room.id]) {
            localMessages = messages[room.id];
          }
          return <RoomWidget key={i}
            topic={room.roomTopic}
            isOpened={room.isOpened}
            isSolved={room.isSolved}
            clientId={id}
            id={room.id}
            role={role}
            sendMessageCallback={this.sendMessageCallback}
            markAsResolvedCallback={this.props.markRoomAsResolved}
            markAsUnresolvedCallback={this.props.markRoomAsUnresolved}
            messages={localMessages} />;
        })}
        <CreateNewRoomWidget
          createRoomCallback={this.createRoomCallback} />
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClientRoomPage);
