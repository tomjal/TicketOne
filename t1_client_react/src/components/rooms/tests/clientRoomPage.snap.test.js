import React from 'react';
import renderer from 'react-test-renderer';

import { ClientRoomPage } from './../clientRoomPage';

test('ClientRoomPage snapshot', () => {
  const props = { getMessagesByRoom: () => null };
  const tree = renderer.create(<ClientRoomPage {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});