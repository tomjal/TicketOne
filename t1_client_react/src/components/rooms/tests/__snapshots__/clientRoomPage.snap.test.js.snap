// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientRoomPage snapshot 1`] = `
<div
  className="room-flex-container"
>
  <RoomWidget
    sendMessageCallback={[Function]}
  />
</div>
`;
